hack/generated/pkg/util/armresourceresolver/resolver.go

```diff
@@ -78,24 +112,49 @@ func (r *Resolver) GetResourceGroupAndFullAzureName(ctx context.Context, obj gen
 
 	ownerObj, err := r.client.GetObject(ctx, ownerNamespacedName, ownerGvk)
 	if err != nil {
-		return "", "", errors.Wrapf(err, "couldn't find owner %s of %s", owner.Name, obj.GetName())
+		if apierrors.IsNotFound(err) {
+			err := NewOwnerNotFoundError(ownerNamespacedName)
+			// TODO: I don't know how to preserve the stack of the existing error here...?
+			return nil, errors.WithStack(err)
+		}
+
+		return nil, errors.Wrapf(err, "couldn't find owner %s of %s", owner.Name, obj.GetName())
 	}
 
 	ownerMeta, ok := ownerObj.(genruntime.MetaObject)
 	if !ok {
-		return "", "", errors.Errorf("owner %s (%s) was not of type genruntime.MetaObject", ownerNamespacedName, ownerGvk)
+		return nil, errors.Errorf("owner %s (%s) was not of type genruntime.MetaObject", ownerNamespacedName, ownerGvk)
 	}
 
-	rgName, ownerName, err := r.GetResourceGroupAndFullAzureName(ctx, ownerMeta)
-	if err != nil {
-		return "", "", errors.Wrapf(err, "failed to get full Azure name and resource group for %s", ownerNamespacedName)
+	return ownerMeta, nil
+}
+
+func (r *Resolver) findGVKOrError(owner *genruntime.ResourceReference) (schema.GroupVersionKind, error) {
+	var ownerGvk schema.GroupVersionKind
+	found := false
+	// TODO: We need to find the specific storage version GVK...
+	for gvk := range r.client.Scheme.AllKnownTypes() {
+		if gvk.Group == owner.Group && gvk.Kind == owner.Kind {
+			if !found {
+				ownerGvk = gvk
+				found = true
+			} else {
+				return ownerGvk, errors.Errorf("owner group: %s, kind: %s has multiple possible schemes registered", owner.Group, owner.Kind)
```

Comment:
I do want to punt on this, but I think for now it's ok leaving this as is.

This needs to get fixed as part of the automation to register all types with the controller automatically, but right now:
1. That automation doesn't exist, we're registering everything manually. For now we can just avoid registering two different versions of the same service in the controller.
2. Automatic registration is blocked behind a few other issues right now, because the actual generated CRDs still aren't 100% valid (recursively defined types and OneOf being on the Kube object instead of the ARM object are both resulting in invalid CRDs that cannot be installed).
3. I'd rather fail catastrophically in this case so that we don't forget to fix it.

Thoughts?

Reply:
Yeah, that makes sense - we'll definitely know as soon as we see this error.